modules/client/src/addressList/internal/client/methods.ts

```diff
@@ -136,6 +128,23 @@ export class ClientAddressListMethods extends ClientCore
       proposalId: hexZeroPad(parsedLog.args["voteId"].toHexString(), 32),
     };
   }
+
+  /**
+   * Pins a metadata object into IPFS and retruns the generated hash
+   *
+   * @param {ProposalMetadata} params
+   * @return {*}  {Promise<string>}
+   * @memberof ClientMethods
+   */
+  public async pinMetadata(params: ProposalMetadata): Promise<string> {
+    try {
+      const cid = await this.ipfs.add(JSON.stringify(params));
+      await this.ipfs.pin(cid);
+      return cid;
```

Comment:
IMO we should return a URI, not the CiD.... or otherwise refactor the rest of the places where this is used, because there are still mismatches